pyiron_base/generic/units.py

```diff
@@ -0,0 +1,300 @@
+
+    >>> base_units = PyironUnitRegistry()
+    >>> base_units.add_quantity(quantity="energy", unit=pint_registry.eV, data_type=float)
+
+    Labels corresponding to a particular physical quantity can also be registered
+
+    >>> base_units.add_labels(labels=["energy_tot", "energy_pot"], quantity="energy")
+
+    For more information on working with `pint`, see: https://pint.readthedocs.io/en/0.10.1/tutorial.html
+
+    """
+    def __init__(self):
+        """
+        Attributes:
+            self.quantity_dict
+            self.unit_dict
+            self.dtype_dict
+        """
+        self._quantity_dict = dict()
+        self._dtype_dict = dict()
+        self._unit_dict = dict()
+
+    @property
+    def quantity_dict(self):
+        """
+        A dictionary of the different labels stored and the physical quantity they correspond to
+
+        Returns:
+            dict
+        """
+        return self._quantity_dict
+
+    @property
+    def dtype_dict(self):
+        """
+        A dictionary of the different physical quantities and the corresponding datatype in which they are to be stored
+
+        Returns:
+            dict
+        """
+        return self._dtype_dict
+
+    @property
+    def unit_dict(self):
+        """
+        A dictionary of the different physical quantities and the corresponding `pint` unit
+
+        Returns:
+            dict
+        """
+        return self._unit_dict
+
+    def add_quantity(self, quantity, unit, data_type=float):
+        """
+        Add a quantity to a registry
+
+        Args:
+            quantity (str): The physical quantity
+            unit (pint.unit.Unit/pint.quantity.Quantity): `pint` unit or quantity
+            data_type (type): Data type in which the quantity has to be stored
+
+        """
+        if not isinstance(unit, (pint.unit.Unit, pint.quantity.Quantity)):
+            raise ValueError("The unit should be a `pint` unit or quantity")
+        self._unit_dict[quantity] = unit
+        self._dtype_dict[quantity] = data_type
+
+    def add_labels(self, labels, quantity):
+        """
+        Maps quantities with different labels to quantities already defined in the registry
+
+        Args:
+            labels (list/ndarray): List of labels
+            quantity (str): Physical quantity associated with the labels
+
+        Note: `quantity` should already be a key of unit_dict
+
+        """
+        for label in labels:
+            if quantity in self.unit_dict.keys():
+                self._quantity_dict[label] = quantity
+            else:
+                raise ValueError("Quantity {} is not defined. Use `add_quantity` to register the unit of this label")
+
+    def __getitem__(self, item):
+        """
+        Getter to return corresponding `pint` unit for a given quantity
+
+        Args:
+            item (str):
+
+        Returns:
+            pint.unit.Unit/pint.quantity.Quantity: The corresponding `pint` unit/quantity
+        """
```

Comment:
```suggestion
Raises:
   ...
        """
```

Reply:
Generally there's a few places where the raised exception are not documented yet.